src/robot/dial.ts

```diff
@@ -0,0 +1,122 @@
+import { RobotClient } from './Client';
+
+interface DialDirectConf {
+  authEntity?: string;
+  host: string;
+  locationSecret?: string;
```

Comment:
how does one dial something that is not viam managed and with other credentials? locationSecret is one such type. Might it be better to more closely match the go options?

Reply:
fair, i did this to make dialing viam more convenient but this is too deeply coupled at this point

Reply:
yeah and we want to make sure we always are supporting our OSS cases well in addition to other types of credentials that an app managed robot can already accept (like a JWT)

Reply:
i was considering letting this function accept a union of interfaces that could support these different use cases, but that maybe that's a bit confusing...